src/main/java/spoon/compiler/Environment.java

```diff
@@ -589,6 +589,16 @@
 	 * @return true if spoon is allowed to create a model of a project that contains multiple copies of the same class
 	 */
 	boolean isIgnoreDuplicateDeclarations();
+	/**
+	 * Whether Spoon currently ignores JLS correctness.
+	 * @return true if spoon ignores JLS correctness, false if spoon throws {@link spoon.JLSViolationException} when it detects JLS violation.
+	 */
+	boolean isIgnoreJLSCorrectness();
+	/**
+	 * Sets whether Spoon should ignore JLS violations.
+	 * @param ignoreJLSCorrectness true if spoon should ignore JLS violations, false if spoon should throw {@link spoon.JLSViolationException} when it detects JLS violation.
+	 */
+	void setIgnoreJLSCorrectness(boolean ignoreJLSCorrectness);
```

Comment:
This concept is extremely close to `setIgnoreSyntaxErrors`. A JLS violation is in most situations a syntax error. If this is to be added as a separate mode, it needs to be clearly distinguished from `setIgnoreSyntaxErrors`.

The key thing about `setIgnoreSyntaxErrors` is that it's a filter to ignore files with syntax errors in the input. This new proposed flag could perhaps distinguish itself by being explicitly about _maintaining_ JLS compliance in the Spoon model.

What I'm trying to get across is that if I were to read these docs, the distinction between this flag and `setIgnoreSyntaxErrors` would not be clear to me.

Finally, I'd argue this should be called "ignore JLS violations" or "skip JLS correctness checks".  I understand what "ignore correctness" is meant to imply, but it's a bit of a brain twister for me.

Reply:
```java
	/**
	 * Sets the option ignore-syntax-errors to remove files with any syntax errors from the compilation batch.
	 *
	 * @param ignoreSyntaxErrors whether Spoon should ignore files with any syntax errors or JLS violations
	 */
	void setIgnoreSyntaxErrors(boolean ignoreSyntaxErrors);
```
After reading these docs, I think there is a problem with merging these 2 flags, because building a model with a wrong identifier is possible, but the flag removes the file completely. I'm unsure if ignoring = removing is expected behavior.